modules/reference/pages/cluster-properties.adoc

```diff
@@ -1,1796 +1,5372 @@
+Enables raft optimization of heartbeats.
+
+*Requires Restart:* No
+
+*Nullable:* No
+
+*Visibility:* `tunable`
+
+*Type:* boolean
+
+*Default:* `true`
+
+---
+
+=== raft_flush_timer_interval_ms
+
+Interval of checking partition against the `raft_replica_max_pending_flush_bytes`.
+
+*Requires Restart:* No
+
+*Nullable:* No
+
+*Visibility:* `tunable`
+
+*Type:* integer
+
+*Accepted values:* [`-17592186044416`, `17592186044415`]
+
+*Default:* `100ms`
+
+---
+
+=== raft_heartbeat_disconnect_failures
+
+After how many failed heartbeats to forcibly close an unresponsive TCP connection.  Set to 0 to disable force disconnection.
+
+*Requires Restart:* No
+
+*Nullable:* No
+
+*Visibility:* `tunable`
+
+*Type:* integer
+
+*Default:* `3`
+
+---
+
+=== raft_heartbeat_interval_ms
+
+Milliseconds for raft leader heartbeats.
+
+*Requires Restart:* No
+
+*Nullable:* No
+
+*Visibility:* `tunable`
+
+*Type:* integer
+
+*Accepted values:* [`-17592186044416`, `17592186044415`]
+
+*Default:* `std::chrono::milliseconds(150)`
+
+---
+
+=== raft_heartbeat_timeout_ms
+
+Raft heartbeat RPC timeout.
+
+*Requires Restart:* No
+
+*Nullable:* No
+
+*Visibility:* `tunable`
+
+*Type:* integer
+
+*Accepted values:* [`-17592186044416`, `17592186044415`]
+
+*Default:* `3s`
+
+---
+
+=== raft_io_timeout_ms
+
+Raft I/O timeout.
+
+*Requires Restart:* No
+
+*Nullable:* No
+
+*Visibility:* `tunable`
+
+*Type:* integer
+
+*Accepted values:* [`-17592186044416`, `17592186044415`]
+
+*Default:* `10000ms`
+
+---
+
+=== raft_learner_recovery_rate
+
+Raft learner recovery rate limit. Throttles the rate of data communicated to nodes (learners) that need to catch up to leaders.
+
+This rate limit is placed on a node sending data to a recovering node. Each sending node will be limited to this rate. The recovering node will accept data as fast as possible according to the combined limits of all healthy nodes in the cluster. For example, if two nodes are sending data to the recovering node, and `raft_learner_recovery_rate` is 100 MB/sec, then the recovering node will recover at a rate of 200 MB/sec.
+
+*Units*: bytes per second.
+
+*Requires Restart:* No
+
+*Nullable:* No
+
+*Visibility:* `tunable`
+
+*Type:* integer
+
+*Default:* `104857600`
+
+---
+
+=== raft_max_concurrent_append_requests_per_follower
+
+Maximum number of concurrent append entries requests sent by leader to one follower.
+
+*Requires Restart:* No
+
+*Nullable:* No
+
+*Visibility:* `tunable`
+
+*Type:* integer
+
+*Accepted values:* [`0`, `4294967295`]
+
+*Default:* `16`
+
+---
+
+=== raft_max_recovery_memory
+
+Max memory that can be used for reads in raft recovery process by default 15% of total memory.
+
+*Requires Restart:* No
+
+*Nullable:* Yes
+
+*Visibility:* `tunable`
+
+*Type:* integer
+
+*Default:* `null`
+
+---
+
+=== raft_recovery_concurrency_per_shard
+
+How many partitions may simultaneously recover data to a particular shard. This is limited to avoid overwhelming nodes when they come back online after an outage.
```

Comment:
```suggestion
How many partitions may simultaneously recover data to a particular shard. This is limited to avoid overwhelming nodes when they come back online after an outage.
```
```suggestion
Use to specify the number of partitions that may simultaneously recover data to a particular shard. This number is limited to avoid overwhelming nodes when they come back online after an outage.
```